### PR TITLE
Add generate-reclaim-csv command

### PR DESCRIPTION
**Pull request by tspascoal**
With a CSV reclaim file users will be able to reclaim users in bulk.

Issue #335 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created) - #360

### CODE REVIEW

File: src/Octoshift/GithubApi.cs [hunk 1]
```diff
@@ -481,32 +481,7 @@ public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
...
-                        }
-                    }
-                }";
-
-            var payload = new
-            {
-                query = $"{query} {{ {gql} }}",
-                variables = new { id = orgId }
-            };
+            var payload = GetMannequinsPayload(orgId);
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:484]: This is only partially DRY. Instead it would be better to call `GetMannequins` method here and then build or query by calling `FirstOrDefaultAsync()`. The only thing here is that `GetMannequins` should return an actual `IEnumerable` without calling `ToListAsync` so that inside `GetMannequin` method we can build our query off of the returned `IEnumerable`. Our method will look like the following after the changes:
```c#
public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
{
    var url = $"{_apiUrl}/graphql";

    var payload = GetMannequinsPayload(orgId);

    var mannequin = await _client
        .GetMannequins(orgId)
        .FirstOrDefaultAsync(jToken => username.Equals((string)jToken["login"], StringComparison.OrdinalIgnoreCase));
    
    return mannequin is null ? new Mannequin() : MannequinBuilder(mannequin);
}
```

File: src/Octoshift/GithubApi.cs [hunk 2]
```diff
@@ -515,20 +490,24 @@ public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
...
+
+            return await _client.PostGraphQLWithPaginationAsync(
+                    url,
+                    payload,
+                    data => (JArray)data["data"]["node"]["mannequins"]["nodes"],
+                    data => (JObject)data["data"]["node"]["mannequins"]["pageInfo"])
+                .Select(mannequin => MannequinBuilder(mannequin)
+
+                )
+                .ToListAsync();
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:510]: Based on my suggestion for `GetMannequin` the `ToListAsync` should be removed so that we will return the actual `IEnumerable` object.

File: src/Octoshift/GithubApi.cs [hunk 3]
```diff
@@ -581,5 +560,51 @@ public virtual async Task<MannequinReclaimResult> ReclaimMannequin(string orgId,
...
+                Login = (string)mannequin["login"],
+                MappedUser = mannequin["claimant"].Any()
+                                    ? new Claimant
+                                    {
+                                        Id = (string)mannequin["claimant"]["id"],
+                                        Login = (string)mannequin["claimant"]["login"]
+                                    }
+                                    : null
+            };
+        }
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:608]: Again based on my suggestion there will be no need to create these private methods.

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class GenerateReclaimCSVCommand : Command
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:10]: Based on C# 's naming guidelines CSV should camel cased --> GenerateReclaimCsvCommand`

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+                IsRequired = false
+            };
+
+            AddOption(githubTargetOrgOption);
+            AddOption(outputOption);
+            AddOption(forceOption);
+            AddOption(verbose);
+
+            Handler = CommandHandler.Create<string, string, bool, bool>(Invoke);
+        }
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:56]: all of our commands now support providing the PATs through args. So here you also need to add an extra (not required) option for provide the GitHub PAT.

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+        public async Task Invoke(
+          string githubTargetOrg,
+          string output,
+          bool includeReclaimed = false,
+          bool verbose = false)
+        {
+            const string targetApiUrl = "https://api.github.com";
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:64]: this needs to be removed since this is the default value of the Target API.

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+            _log.Verbose = verbose;
+
+            _log.LogInformation("Generating CSV...");
+
+            _log.LogInformation($"GITHUB TARGET ORG: {githubTargetOrg}");
+            _log.LogInformation($"FILE: {output}");
+            _log.LogInformation($"INCLUDE RECLAIMED: {includeReclaimed}");
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:72]: for all non required args we only needs to log them if they are provided:
```c#
if (includeReclaimed is not null)
{
   ...
}
```

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+            _log.LogInformation($"    # Mannequins Found: {mannequins.Count()}");
+
+            var numberMannequins = 0;
+            var contents = "login,claimantlogin\n";
+            foreach (var mannequin in mannequins)
+            {
+                if (mannequin.MappedUser != null && !includeReclaimed)
+                {
+                    continue;
+                }
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:90]: I'd convert this to a `Where` clause:
```c#
foreach (var mannequin in mannequins.Where(m => includeReclaimed || m.MappedUser is null))
{
    ...
}
```

File: src/gei/Commands/GenerateReclaimCSVCommand.cs [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                numberMannequins++;
+                contents += $"{mannequin.Login},{mannequin.MappedUser?.Login}\n";
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCSVCommand.cs:92]: instead of appending to a string use a `StringBuilder` instead. CodeQL also warns about this.

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+using OctoshiftCLI.GithubEnterpriseImporter;
+using OctoshiftCLI.GithubEnterpriseImporter.Commands;
+using OctoshiftCLI.Models;
+using Xunit;
+
+namespace OctoshiftCLI.Tests.GithubEnterpriseImporter.Commands
+{
+    public class GenerateReclaimCSVCommandTests
+    {
+        private const string TARGET_API_URL = "https://api.github.com";
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:14]: no need to set this since it is the default API url.

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+            Assert.Equal(4, command.Options.Count);
+
+            TestHelpers.VerifyCommandOption(command.Options, "github-target-org", true);
+            TestHelpers.VerifyCommandOption(command.Options, "output", true);
+            TestHelpers.VerifyCommandOption(command.Options, "include-reclaimed", false);
+            TestHelpers.VerifyCommandOption(command.Options, "verbose", false);
+        }
+
+        [Fact]
+        public async void NoMannequins_GenerateEmptyCSV_WithOnlyHeaders()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:31]: CSV --> Csv

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+            // Act
+            await command.Invoke("octocat", "unit-test-output", false);
+
+            // Assert
+            csvContent.Should().Be("login,claimantlogin\n");
+        }
+
+        [Fact]
+        public async void Mannequins_GenerateCSV_UnreclaimedOnly()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:65]: `async void` does not make any sense. It should be `async Task` instead. otherwise exceptions won't be surfaced properly and will be swallowed.

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs (hunk 11 above)
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:31]: `async Task`

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+            // Act
+            await command.Invoke(githubOrg, "unit-test-output", false);
+
+            // Assert
+            csvContent.Should().Be("login,claimantlogin\nmona,\n");
+            mockGithubApi.Verify(x => x.GetMannequins(githubOrgId));
+        }
+
+        [Fact]
+        public async void Mannequins_GenerateCSV_IncludeAlreadyReclaimed()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:120]: `async Task`

**Review by ArinGhazarian — CHANGES_REQUESTED** [review]

File: src/gei/Commands/GenerateReclaimCSVCommand.cs (hunk 4 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateReclaimCSVCommand.cs]: Done [896169](https://github.com/github/gh-gei/pull/365/commits/8961698de8f74966b45b04cbc31baee33cd9f756)

File: src/gei/Commands/GenerateReclaimCSVCommand.cs (hunk 9 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateReclaimCSVCommand.cs]: done [709327b](https://github.com/github/gh-gei/pull/365/commits/709327ba7c16adcdd261313fe74870c903a47116)

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs (hunk 10 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:14]: done

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+            TestHelpers.VerifyCommandOption(command.Options, "github-target-org", true);
+            TestHelpers.VerifyCommandOption(command.Options, "output", true);
+            TestHelpers.VerifyCommandOption(command.Options, "include-reclaimed", false);
+            TestHelpers.VerifyCommandOption(command.Options, "verbose", false);
+        }
+
+        [Fact]
+        public async void NoMannequins_GenerateEmptyCSV_WithOnlyHeaders()
+        {
+            var githubOrg = "FooOrg";
```
[the remark this answers is not in the document]
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:33]: done

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs (hunk 12 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:67]: done

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs (hunk 11 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:31]: done

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs (hunk 13 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateReclaimCSVCommandTests.cs:125]: done

File: src/gei/Commands/GenerateReclaimCSVCommand.cs (hunk 7 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateReclaimCSVCommand.cs:72]: includeReclaimed is a value type it cannot be null. What I'm missing?

changed to
```
if (includeReclaimed)
            {
                _log.LogInformation($"INCLUDING RECLAIMED");
            }
```

File: src/gei/Commands/GenerateReclaimCSVCommand.cs (hunk 5 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateReclaimCSVCommand.cs]: done [3b1aaf](https://github.com/github/gh-gei/pull/365/commits/3b1aaf711d4b9f5e01bbd3750bbe2abbedd62492)

File: src/gei/Commands/GenerateReclaimCSVCommand.cs (hunk 7 above)
Comment (ArinGhazarian) [reply src/gei/Commands/GenerateReclaimCSVCommand.cs]: oh yeah sure, my bad 😃

File: src/Octoshift/AzureApi.cs [hunk 15]
```diff
@@ -49,14 +49,14 @@ private Uri GetServiceSasUriForBlob(BlobClient blobClient)
...
             {
                 BlobContainerName = blobClient.GetParentBlobContainerClient().Name,
                 BlobName = blobClient.Name,
-                Resource = "b"  // Resource = "b" for blobs, "c" for containers
-            };
+                Resource = "b",  // Resource = "b" for blobs, "c" for containers
 
-            sasBuilder.ExpiresOn = DateTimeOffset.UtcNow.AddHours(AUTHORIZATION_TIMEOUT_IN_HOURS);
+                ExpiresOn = DateTimeOffset.UtcNow.AddHours(AUTHORIZATION_TIMEOUT_IN_HOURS)
+            };
```
Comment (ArinGhazarian) [inline src/Octoshift/AzureApi.cs:59]: I wouldn't do this refactor as part of this PR.

File: src/gei/Commands/GenerateReclaimCsvCommand.cs [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+
+        private readonly OctoLogger _log;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+
+        public GenerateReclaimCsvCommand(OctoLogger log, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider) : base("generate-reclaim-csv")
+        {
+            _log = log;
+            _targetGithubApiFactory = targetGithubApiFactory;
+            _environmentVariableProvider = environmentVariableProvider;
```
Comment (ArinGhazarian) [inline src/gei/Commands/GenerateReclaimCsvCommand.cs:23]: It seems like you're not using the `environmentVariableProvider` so it can be removed from the `ctor`

File: src/Octoshift/AzureApi.cs (hunk 15 above)
Comment (tspascoal) [reply src/Octoshift/AzureApi.cs]: I wouldn't either. This was probably some formatting that happened automatically and I didn't notice it.

Good catch

Comment (tspascoal) [reply src/Octoshift/AzureApi.cs]: I wouldn't either. This was probably some formatting that happened automatically and I didn't notice it.

File: src/gei/Commands/GenerateReclaimCsvCommand.cs (hunk 16 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateReclaimCsvCommand.cs:23]: deleted

File: src/Octoshift/GithubApi.cs (hunk 1 above)
Comment (ArinGhazarian) [reply src/Octoshift/GithubApi.cs:484]: Based on our Zoom sync up, for simplicity we decided that it would be better not to go ahead with my proposed changes for this since `GetMannequins` has to return an `IAsyncEnumerable` and it is going to make things unnecessarily complicated.

File: src/Octoshift/GithubApi.cs [hunk 17]
```diff
@@ -515,20 +490,24 @@ public virtual async Task<Mannequin> GetMannequin(string orgId, string username)
...
+
+            var payload = GetMannequinsPayload(orgId);
+
+            return await _client.PostGraphQLWithPaginationAsync(
+                    url,
+                    payload,
+                    data => (JArray)data["data"]["node"]["mannequins"]["nodes"],
+                    data => (JObject)data["data"]["node"]["mannequins"]["pageInfo"])
+                .Select(mannequin => MannequinBuilder(mannequin)
+
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:508]: nit: remove the new line

Comment (tspascoal) [reply src/Octoshift/GithubApi.cs]: done

File: src/Octoshift/GithubApi.cs [hunk 18]
```diff
@@ -581,5 +558,51 @@ public virtual async Task<MannequinReclaimResult> ReclaimMannequin(string orgId,
...
+                }";
+
+            return new
+            {
+                query = $"{query} {{ {gql} }}",
+                variables = new { id = orgId }
+            };
+        }
+
+        private static Mannequin MannequinBuilder(JToken mannequin)
```
Comment (ArinGhazarian) [inline src/Octoshift/GithubApi.cs:592]: nit: a small naming refactor to comply with MS naming guidelines --> `BuildMannequin`

**Review by ArinGhazarian — APPROVED** [review]
Just a small nit but the rest looks good to me 🥇

**Comment by dylan-smith** [comment]
Since we don't have the feature implemented yet to consume this csv, my suggestion is to make this command hidden until we add in the feature that can consume it to claim mannequins in bulk (IsHidden = true on the command).

File: src/Octoshift/GithubApi.cs (hunk 18 above)
Comment (tspascoal) [reply src/Octoshift/GithubApi.cs]: done

**Comment by tspascoal** [comment]
> Since we don't have the feature implemented yet to consume this csv, my suggestion is to make this command hidden until we add in the feature that can consume it to claim mannequins in bulk (IsHidden = true on the command).

Done

**Review by ArinGhazarian — APPROVED** [review]
approved_1

File: RELEASENOTES.md [hunk 19]
```diff
@@ -1,2 +1,3 @@
 - `integrate-boards` no longer requires a PAT with the `All Accessible Organizations` setting
-- Fixed incorrect repo url in migration logs when migrating from GHES
\ No newline at end of file
+- Fixed incorrect repo url in migration logs when migrating from GHES
+- Add `generate-mannequin-csv` command to generate a CSV to be used to reclaim mannequins in bulk.
```
Comment (dylan-smith) [inline RELEASENOTES.md:3]: lets leave this out of the release notes until we remove the IsHidden flag

File: src/Octoshift/AzureApi.cs [hunk 20]
```diff
@@ -49,14 +49,14 @@ private Uri GetServiceSasUriForBlob(BlobClient blobClient)
                 throw new InvalidOperationException("BlobClient object has not been authorized to generate shared key credentials. Verify --azure-storage-connection-key is valid and has proper permissions.");
             }
 
-            var sasBuilder = new BlobSasBuilder()
+            var sasBuilder = new BlobSasBuilder
```
Comment (dylan-smith) [inline src/Octoshift/AzureApi.cs:52]: weren't you going to revert this change?

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+
+        private readonly OctoLogger _log;
+        private readonly GithubApiFactory _githubApiFactory;
+
+        public GenerateMannequinCsvCommand(OctoLogger log, GithubApiFactory githubApiFactory) : base("generate-mannequin-csv")
+        {
+            _log = log;
+            _githubApiFactory = githubApiFactory;
+
+            Description = "Generates a CSV with unreclained mannequins to reclaim them in bulk.";
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:23]:
```suggestion
            Description = "Generates a CSV with unreclaimed mannequins to reclaim them in bulk.";
```

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+
+            var githubOrgOption = new Option<string>("--github-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable."
+            };
+
+            var outputOption = new Option<string>("--output")
+            {
+                IsRequired = true,
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:34]: Can we default this to something (e.g. mannequins.csv) and make it optional arg (like we do with the --output in generate-script command)

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+                Description = "Output filename"
+            };
+
+            var forceOption = new Option("--include-reclaimed")
+            {
+                IsRequired = false,
+                Description = "Map the user even if it was previously mapped"
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:41]: Should this say something like "Include mannequins that have already been reclaimed"

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 24]
```diff
@@ -0,0 +1,107 @@
...
+                Description = "Uses GH_PAT env variable."
+            };
+
+            var outputOption = new Option<string>("--output")
+            {
+                IsRequired = true,
+                Description = "Output filename"
+            };
+
+            var forceOption = new Option("--include-reclaimed")
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:38]:
```suggestion
            var includeReclaimedOption = new Option("--include-reclaimed")
```

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 25]
```diff
@@ -0,0 +1,107 @@
...
+            _log.LogInformation($"FILE: {output}");
+            if (includeReclaimed)
+            {
+                _log.LogInformation("INCLUDING RECLAIMED");
+            }
+
+            var githubApi = _githubApiFactory.Create(personalAccessToken: githubPat);
+
+            var githubOrgId = await githubApi.GetOrganizationId(githubOrg);
+            _log.LogInformation($"    Organization Id: {githubOrgId}");
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:88]: do we need to log this? I don't think your typical user will care about their org id, unless I'm missing something. (it will be in the verbose logs already from the previous API call)

File: src/gei/Commands/GenerateMannequinCsvCommand.cs [hunk 26]
```diff
@@ -0,0 +1,107 @@
...
+
+        private readonly OctoLogger _log;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+
+        public GenerateMannequinCsvCommand(OctoLogger log, ITargetGithubApiFactory targetGithubApiFactory) : base("generate-mannequin-csv")
+        {
+            _log = log;
+            _targetGithubApiFactory = targetGithubApiFactory;
+
+            Description = "Generates a CSV with unreclained mannequins to reclaim them in bulk.";
```
Comment (dylan-smith) [inline src/gei/Commands/GenerateMannequinCsvCommand.cs:23]:
```suggestion
            Description = "Generates a CSV with unreclaimed mannequins to reclaim them in bulk.";
```

File: src/gei/Commands/GenerateMannequinCsvCommand.cs [hunk 27]
```diff
@@ -0,0 +1,107 @@
...
+
+            var githubTargetOrgOption = new Option<string>("--github-target-org")
+            {
+                IsRequired = true,
+                Description = "Uses GH_PAT env variable."
+            };
+
+            var outputOption = new Option<string>("--output")
+            {
+                IsRequired = true,
```
Comment (dylan-smith) [inline src/gei/Commands/GenerateMannequinCsvCommand.cs:34]: same suggestion as above (optional with default)

File: src/gei/Commands/GenerateMannequinCsvCommand.cs [hunk 28]
```diff
@@ -0,0 +1,107 @@
...
+                Description = "Output filename"
+            };
+
+            var forceOption = new Option("--include-reclaimed")
+            {
+                IsRequired = false,
+                Description = "Map the user even if it was previously mapped"
```
Comment (dylan-smith) [inline src/gei/Commands/GenerateMannequinCsvCommand.cs:41]: same wording suggestion as above

File: src/gei/Commands/GenerateMannequinCsvCommand.cs [hunk 29]
```diff
@@ -0,0 +1,107 @@
...
+                Description = "Uses GH_PAT env variable."
+            };
+
+            var outputOption = new Option<string>("--output")
+            {
+                IsRequired = true,
+                Description = "Output filename"
+            };
+
+            var forceOption = new Option("--include-reclaimed")
```
Comment (dylan-smith) [inline src/gei/Commands/GenerateMannequinCsvCommand.cs:38]: same variable name suggestion as above

File: src/gei/Commands/GenerateMannequinCsvCommand.cs [hunk 30]
```diff
@@ -0,0 +1,107 @@
...
+            _log.LogInformation($"FILE: {output}");
+            if (includeReclaimed)
+            {
+                _log.LogInformation("INCLUDING RECLAIMED");
+            }
+
+            var githubApi = _targetGithubApiFactory.Create(targetPersonalAccessToken: githubTargetPat);
+
+            var githubOrgId = await githubApi.GetOrganizationId(githubTargetOrg);
+            _log.LogInformation($"    Organization Id: {githubOrgId}");
```
Comment (dylan-smith) [inline src/gei/Commands/GenerateMannequinCsvCommand.cs:88]: same as above

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs [hunk 31]
```diff
@@ -0,0 +1,180 @@
...
+                    Login = "monalisa",
+                    MappedUser = new Claimant
+                    {
+                        Id = "monalisamapped-id",
+                        Login = "monalisa_gh"
+                    }
+                }
+            };
+
+
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs:89]:
```suggestion
```

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs [hunk 32]
```diff
@@ -0,0 +1,180 @@
...
+            var expected = "login,claimantlogin" + Environment.NewLine
+                + "mona," + Environment.NewLine;
+
+            // Act
+            await command.Invoke(githubOrg, "unit-test-output", false);
+
+            // Assert
+            csvContent.Should().Be(expected);
+            mockGithubApi.Verify(x => x.GetMannequins(githubOrgId));
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs:119]: we usually don't call verify in situations like this, the only way it could have gotten the expected result is to call getmannequins with the proper orgid, so not really any need to call verify.

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs [hunk 33]
```diff
@@ -0,0 +1,180 @@
...
+                + "mona," + Environment.NewLine
+                + "monalisa,monalisa_gh" + Environment.NewLine;
+
+            // Act
+            await command.Invoke(githubOrg, "unit-test-output", true);
+
+            // Assert
+            csvContent.Should().Be(expected);
+
+            mockGithubApi.Verify(x => x.GetMannequins(githubOrgId));
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs:177]:
```suggestion
```

File: src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs [hunk 34]
```diff
@@ -0,0 +1,180 @@
...
+
+            var expected = "login,claimantlogin" + Environment.NewLine
+                + "mona," + Environment.NewLine;
+
+            // Act
+            await command.Invoke(githubOrg, "unit-test-output", false);
+
+            // Assert
+            csvContent.Should().Be(expected);
+            mockGithubApi.Verify(x => x.GetMannequins(githubOrgId));
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs:119]:
```suggestion
```

File: src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs [hunk 35]
```diff
@@ -0,0 +1,180 @@
...
+                + "mona," + Environment.NewLine
+                + "monalisa,monalisa_gh" + Environment.NewLine;
+
+            // Act
+            await command.Invoke(githubOrg, "unit-test-output", true);
+
+            // Assert
+            csvContent.Should().Be(expected);
+
+            mockGithubApi.Verify(x => x.GetMannequins(githubOrgId));
```
Comment (dylan-smith) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs:177]:
```suggestion
```

**Review by dylan-smith — CHANGES_REQUESTED** [review]

File: src/Octoshift/AzureApi.cs (hunk 20 above)
Comment (tspascoal) [reply src/Octoshift/AzureApi.cs:52]: I did.....

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs (hunk 22 above)
Comment (tspascoal) [reply src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:34]: done

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs (hunk 24 above)
Comment (tspascoal) [reply src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:38]: fixed

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs (hunk 25 above)
Comment (tspascoal) [reply src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:88]: removed

File: src/gei/Commands/GenerateMannequinCsvCommand.cs (hunk 26 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateMannequinCsvCommand.cs:23]: fixed

File: src/gei/Commands/GenerateMannequinCsvCommand.cs (hunk 27 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateMannequinCsvCommand.cs:34]: done

File: RELEASENOTES.md (hunk 19 above)
Comment (tspascoal) [reply RELEASENOTES.md]: done

File: src/gei/Commands/GenerateMannequinCsvCommand.cs (hunk 29 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateMannequinCsvCommand.cs:38]: fixed

File: src/gei/Commands/GenerateMannequinCsvCommand.cs (hunk 30 above)
Comment (tspascoal) [reply src/gei/Commands/GenerateMannequinCsvCommand.cs:88]: fixed

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs (hunk 31 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs:89]: fixed

File: src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs (hunk 35 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs:177]: fixed

File: src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs (hunk 34 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs:119]: fixed

File: src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs (hunk 35 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/GenerateReclaimCsvCommandTests.cs:177]: fixed

File: src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs (hunk 33 above)
Comment (tspascoal) [reply src/OctoshiftCLI.Tests/gei/Commands/GenerateMannequinCsvCommandTests.cs:177]: fixed

File: RELEASENOTES.md [hunk 36]
```diff
@@ -1 +1,2 @@
-
+- `integrate-boards` no longer requires a PAT with the `All Accessible Organizations` setting
+- Fixed incorrect repo url in migration logs when migrating from GHES
```
Comment (dylan-smith) [inline RELEASENOTES.md:2]:
```suggestion
```

I think you merged this incorrectly

File: src/ado2gh/Commands/GenerateMannequinCsvCommand.cs [hunk 37]
```diff
@@ -0,0 +1,107 @@
...
+
+            var numberMannequins = 0;
+            var contents = new StringBuilder().AppendLine("login,claimantlogin");
+            foreach (var mannequin in mannequins.Where(m => includeReclaimed || m.MappedUser is null))
+            {
+                numberMannequins++;
+                contents.AppendLine($"{mannequin.Login},{mannequin.MappedUser?.Login}");
+            }
+
+            await WriteToFile(output, contents.ToString());
```
Comment (dylan-smith) [inline src/ado2gh/Commands/GenerateMannequinCsvCommand.cs:102]:
```suggestion
            await WriteToFile(output.FullName, contents.ToString());
```

If we change it to FileInfo I think that's what we need to do to get the path

**Review by dylan-smith — APPROVED** [review]
